src/aggregation/bucket/filter.rs

```diff
@@ -0,0 +1,1758 @@
+}
+
+// Enable serialization of SerializableQuery trait objects using erased-serde
+erased_serde::serialize_trait_object!(SerializableQuery);
+
+/// Filter aggregation creates a single bucket containing documents that match a query.
+///
+/// # Usage
+/// ```rust
+/// use tantivy::aggregation::bucket::SerializableQuery;
+/// use tantivy::aggregation::bucket::filter::FilterAggregation;
+/// use tantivy::query::{Query, EnableScoring, TermQuery, Weight};
+///
+/// #[derive(Debug, Clone)]
+/// struct SerializableTermQuery(TermQuery);
+/// impl SerializableQuery for SerializableTermQuery {
+///     fn clone_box(&self) -> Box<dyn SerializableQuery> {
+///         Box::new(self.clone())
+///     }
+/// }
+///
+/// impl Query for SerializableTermQuery {
+///     fn weight(&self, enable_scoring: EnableScoring<'_>) -> tantivy::Result<Box<dyn Weight>> {
+///         self.0.weight(enable_scoring)
+///     }
+/// }
+///
+/// impl SerializableTermQuery {
+///     pub fn new(term_query: TermQuery) -> Self {
+///         Self(term_query)
+///     }
+/// }
+///
+/// impl serde::Serialize for SerializableTermQuery {
+///     fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+///     where S: serde::Serializer {
+///         "todo".serialize(serializer)
+///     }
+/// }
+///
+/// // Query strings are parsed using Tantivy's standard QueryParser
+/// let filter_agg = FilterAggregation::new("category:electronics AND price:[100 TO 500]".to_string());
+///
+/// // Direct Query objects can be used for custom query types
+/// let term_query = TermQuery::new(
+///     tantivy::Term::from_field_text(tantivy::schema::Field::from_field_id(0), "electronics"),
+///     tantivy::schema::IndexRecordOption::Basic
+/// );
+///
+/// let filter_agg = FilterAggregation::new_with_query(Box::new(SerializableTermQuery::new(term_query)));
+/// ```
+///
+/// # Result
+/// The filter aggregation returns a single bucket with:
+/// - `doc_count`: Number of documents matching the filter
+/// - Sub-aggregation results computed on the filtered document set
+#[derive(Debug, Clone)]
+pub struct FilterAggregation {
+    /// The query for filtering - can be either a query string or a direct Query object
+    query: FilterQuery,
+}
+
+/// Represents different ways to specify a filter query
+#[derive(Debug)]
+pub enum FilterQuery {
+    /// Query string that will be parsed using Tantivy's standard parsing facilities
+    /// Accepts query strings that can be parsed by QueryParser::parse_query()
+    QueryString(String),
+
+    /// Custom Query object for programmatic query construction
+    ///
+    /// This variant allows passing pre-constructed Query objects directly,
+    /// which is useful for:
+    /// - Custom query types not expressible as query strings
+    /// - Programmatic query construction
+    /// - Extension query types
+    ///
+    /// Note: This variant cannot be serialized to JSON (only QueryString can be serialized)
+    CustomQuery(Box<dyn SerializableQuery>),
```

Comment:
Why do we use `SerializableQuery` when the query cannot be serialized?

Reply:
Good suggestion to use a query constructor/builder. Fixed it in [4b94636](https://github.com/quickwit-oss/tantivy/pull/2711/commits/4b94636dc23733b5e1cf7ac19fde7a10a014761d)